Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Nodes.php

```diff
@@ -194,4 +197,12 @@ public function nextAll(Node $referenceNode): self
 
         return new self(array_slice($this->nodes, $referenceNodeIndex + 1));
     }
+
+    public function getNodeAggregateIds(): NodeAggregateIds
+    {
+        return NodeAggregateIds::create(...array_map(
+            fn (Node $node): NodeAggregateId => $node->nodeAggregateId,
+            $this->nodes
+        ));
+    }
```

Comment:
Just a nitpick, but I tend to add more generic `map` and (if needed) `filter` methods to my collection types like:
```suggestion
    /**
      * @param \Closure(Node $node): mixed $callback
      * @return array<mixed>
      **/
    public function map(\Closure $callback): NodeAggregateIds
    {
        return array_map($callback, $this->nodes);
        return NodeAggregateIds::create(...array_map(
            fn (Node $node): NodeAggregateId => $node->nodeAggregateId,
            $this->nodes
        ));
    }
```

With that, we could write:

```php
NodeAggregateIds::fromArray($nodes->map(fn (Node $n) => $node-> nodeAggregateId));
```

but use it for other mappings as well.
We could still keep `getNodeAggregateIds()` for convenience..

Anyways, just a mini nitpick ofc